azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageTypeServlet.java

```diff
@@ -74,10 +82,36 @@ protected void handleGet(
       throws ServletException, IOException {
     try {
       final Map<String, String> templateVariableToValue = new HashMap<>();
-      if (IMAGE_TYPE_WITH_ID_URI_TEMPLATE.match(req.getRequestURI(), templateVariableToValue)) {
+      ImageTypeDTO imageTypeDTO = null;
+      if (StringUtils.isNotBlank(req.getPathInfo())) {
+        String imageTypeName = req.getPathInfo().substring(INDEX_AFTER_LEADING_SLASH);
+        if (imageTypeName.contains(PATH_SEPARATOR)) {
+          throw new ImageMgmtInvalidInputException(
+              "The path provided is not supported by this API. Please check the documentation.");
+        }
+        if (hasImageManagementPermission(imageTypeName, session.getUser(), Type.GET)) {
```

Comment:
Cases should be like:
1. Proxy user is provided: Validate the logged in user has permission for the image type and proxy user also has permission for the image type (Proxy user can be either Azkaban admin or image type owner.)
2. Proxy user is not provided: Check logged in user has permission to get the image type. Validate from Azkaban admin side as well as image type ownership.
If proxy user is specified and logged in user has permission then it will not go in else if case which is wrong.

Reply:
Forgive my lack of context but could you provide an example scenario where the additional proxy user parameter needs to be supported? What purpose is it serving?  
Why can't we use headless accounts to make the API call? In that case the logged in user is the headless account and we just need to verify permissions on one user, as opposed to logging in as ME providing proxy-user=XYZ where 2 user permissions would need to be checked.

Reply:
Forgive my lack of context but could you provide an example scenario where the additional proxy user parameter needs to be supported? What purpose is it serving?
Why can't we use headless accounts to make the API call? In that case the logged in user is the headless account and we just need to verify permissions on one user, as opposed to logging in as ME providing proxy-user=XYZ where 2 user permissions would need to be checked.

Reply:
@ypadron-in I have shared answer for this question in another thread with "Logic will be like:" as starting. Please let me know if you have any other concerns.